integration_tests/test_register.py

```diff
@@ -0,0 +1,51 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import uuid
+
+import integration_tests
+from snapcraft.tests import fixture_setup
+
+
+class RegisterTestCase(integration_tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.useFixture(fixture_setup.StagingStore())
+
+    # FIXME: The store doesn't provide a way to unregister a name *and*
+    # registrations are rate-limited for a given user. We work around that by
+    # creating unique names and assuming we only run against staging or local
+    # dev instances -- vila 2016-04-08
```

Comment:
No. And I don't think there will be one implemented in the foreseeable future (see above for some complexity associated with snap names).
I put this FIXME there to explain why the issue is tricky from a test pov, this is the boundary blocking other type of tests by having to use a staging server.
The long term plan would be to have more tests around register-name against a dev server where we can more easily use scratch DBs and don't have to care about cleaning nor having to use shared resources with uuid which a pita for test writers.

But until we get there, tests against staging cover most of our needs.